packages/react-components/react-icons-compat/src/icon.ts

```diff
@@ -0,0 +1,222 @@
+   * registered, because this scenario can create unexpected consequences.
+   *
+   * Some scenarios include:
+   *
+   * Icon set was previously registered using a different base url.
+   * Icon set was previously registered but a different version was provided.
+   * Icons in a previous registered set overlap with a new set.
+   *
+   * To simply ignore previously registered icons, you can specify to disable warnings. This means
+   * that if an icon which was previous registered is registered again, it will be silently ignored.
+   * However, consider whether the problems listed above will cause issues.
+   **/
+  disableWarnings: boolean;
+
+  /**
+   * @deprecated Use `disableWarnings` instead.
+   */
+  warnOnMissingIcons?: boolean;
+}
+
+export interface IIconRecords {
+  __options: IIconOptions;
+  __remapped: { [key: string]: string };
+  [key: string]: IIconRecord | {};
+}
+
+const ICON_SETTING_NAME = 'icons';
+
+const _iconSettings = GlobalSettings.getValue<IIconRecords>(ICON_SETTING_NAME, {
+  __options: {
+    disableWarnings: false,
+    warnOnMissingIcons: true,
+  },
+  __remapped: {},
+});
+
+/**
+ * Normalizes an icon name for consistent mapping.
+ * Current implementation is to convert the icon name to lower case.
+ *
+ * @param name - Icon name to normalize.
+ * @returns {string} Normalized icon name to use for indexing and mapping.
+ */
+const normalizeIconName = (name: string): string => name.toLowerCase();
+
+/**
+ * Registers a given subset of icons.
+ *
+ * @param iconSubset - the icon subset definition.
+ */
+export function registerIcons(iconSubset: IIconSubset, options?: Partial<IIconOptions>): void {
+  let subset = {
+    ...iconSubset,
+    isRegistered: false,
+    className: undefined,
+  };
+  let { icons } = iconSubset;
+
+  // Grab options, optionally mix user provided ones on top.
+  options = options ? { ..._iconSettings.__options, ...options } : _iconSettings.__options;
+
+  for (const iconName in icons) {
+    if (icons.hasOwnProperty(iconName)) {
+      const code = icons[iconName];
+      const normalizedIconName = normalizeIconName(iconName);
+
+      if (_iconSettings[normalizedIconName]) {
+        _warnDuplicateIcon(iconName);
+      } else {
+        _iconSettings[normalizedIconName] = {
+          code,
+          subset,
+        } as IIconRecord;
+      }
+    }
+  }
+}
+
+/**
+ * Unregisters icons by name.
+ *
+ * @param iconNames - List of icons to unregister.
+ */
+export function unregisterIcons(iconNames: string[]): void {
+  const options = _iconSettings.__options;
+
+  for (const iconName of iconNames) {
+    const normalizedIconName = normalizeIconName(iconName);
+    if (_iconSettings[normalizedIconName]) {
+      delete _iconSettings[normalizedIconName];
+    } else {
+      // Warn that we are trying to delete an icon that doesn't exist
+      if (!options.disableWarnings) {
+        warn(`The icon "${iconName}" tried to unregister but was not registered.`);
```

Comment:
I would suggest just using `console.warn` directly so you don't have to port the `warn` functionality for now.

```suggestion
        console.warn(`The icon "${iconName}" tried to unregister but was not registered.`);
```